src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, Drawable {
+    private String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void draw() {
+        System.out.printf("Figure: %s, area: %.2f sq.units",
+                getFigureName(), getArea());
+    }
+
+    private String getFigureName() {
+        return String.join(" ",
+                getClass().getSimpleName().split("(?=\\p{Upper})")).toLowerCase();
+    }
```

Comment:
Do you really need these methods? You can already do this in an overridden method in shapes

Reply:
Please, look at this peace of code one more time. I 've created getFigureMethod in Abstract class and all other subclasses just extend this method and add only their new options. But if you think it would be better to write this peace of code in each code - I'll changed it.